Реализация http клиента

SpaceCowboy1 opened the pull request.

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

spaceCowboy11 committed:
Реализация http клиента

avfyodorov commented on src/http/handlers/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+    protected final Gson gson = new GsonBuilder().registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter()).create();
+    protected TaskManager taskManager;
+    protected HistoryManager historyManager;
+    String response;
```
Для соблюдения инкапсуляции переменную лучше сделать приватной.

avfyodorov commented on test/HttpTest.java:
```diff
@@ -0,0 +1,176 @@
+
+public class HttpTest {
+    TaskManager manager = new InMemoryTaskManager();
+    HistoryManager historyManager = new InMemoryHistoryManager();
```
Менеджер истории лучше получать через соответствующий метод класса Managers.

avfyodorov commented on test/HttpTest.java:
```diff
@@ -0,0 +1,176 @@
+    TaskManager manager = new InMemoryTaskManager();
+    HistoryManager historyManager = new InMemoryHistoryManager();
+    HttpTaskServer taskServer = new HttpTaskServer(manager, historyManager);
+    Gson gson = new GsonBuilder().registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter()).create();
```
Создание gson также можно было бы поместить в класс Managers, чтобы он был один на весь проект.

avfyodorov commented on src/http/handlers/BaseHttpHandler.java:
```diff
@@ -0,0 +1,48 @@
+public class BaseHttpHandler implements HttpHandler {
+
+    protected static final Charset CHARSET = StandardCharsets.UTF_8;
+    protected final Gson gson = new GsonBuilder().registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter()).create();
```
Создание gson также можно было бы поместить в класс Managers, чтобы он был один на весь проект.

avfyodorov commented on test/HttpTest.java:
```diff
@@ -0,0 +1,176 @@
+    public void shutDown() {
+        taskServer.stop();
+    }
+
```
Хорошо бы и для подзадач тоже добавить тесты.

avfyodorov reviewed (commented):
Добрый вечер, Даниил! 
В целом очень хорошая работа, от меня только несколько уточнений.

spaceCowboy11 committed:
Добавлен модификатор доступа переменной response
Добавлен вызов менеджера через Managers
Добавлено создание Json через Managers
Добавлены тесты Subtask

avfyodorov reviewed (commented):
Добрый день, Даниил!

Замечаний нет.
Работа принята.

Есть просьба-оставлять хоть какую-то реакцию на мои комментарии. Сделано/ Не сделано, потому что не согласен/ Не заметил. 🙂
